frontend/src/components/button/button.tsx

```diff
@@ -0,0 +1,68 @@
+import { type ReactNode } from 'react';
+
+type ButtonVariant = 'primary' | 'secondary' | 'tertiary';
+type ButtonSize = 'small' | 'medium';
+type ButtonType = 'button' | 'submit';
+
+type ButtonProperties = {
+    children: ReactNode;
+    size: ButtonSize;
+    variant: ButtonVariant;
+    type?: ButtonType;
+    disabled?: boolean;
+    className?: string;
+    onClick?: () => void;
+};
+
+const classCreator = (input: string): string =>
+    input
+        .replaceAll(/\s+/gm, ' ')
+        .split(' ')
+        .filter((item) => typeof item === 'string')
+        .join(' ')
+        .trim();
```

Comment:
I think it would be better if we move this into separate helper and use clsx or classnames packages for it.
Since we would use this helper in other components